grakn/service/Session/util/Communicator.py

```diff
@@ -85,13 +85,15 @@ def __init__(self, communicator, request, is_last_response):
         self._is_last_response = is_last_response
         self._communicator = communicator
         self._ended = False
+        self._started = False
         self._response_buffer = deque()
         communicator._send_with_resolver(self, request)
 
     def __iter__(self):
         return self
 
     def __next__(self):
+        self._started = True
```

Comment:
is this required if it's also in `_buffer_response()`?

Reply:
Yes, `_buffer_response()` is called when the result is received before the client attempts to get it. `__next__()` may be called first, in which case it blocks and a corresponding call to `_buffer_response()` is not made.

Reply:
To be more accurate, `_buffer_response()` is actually called when you attempt to receive a result that comes after the result of a previous request, and so you must buffer those results to the corresponding resolver.